### PR TITLE
feat: Add parallelism to argo-helm chart

### PR DESCRIPTION
**Pull request by dbseel**
https://github.com/argoproj/argo-helm/issues/476

Checklist:

* [x] I have update the chart version in `Chart.yaml` following Semantic Versioning.
* [x] Any new values are backwards compatible and/or have sensible default.
* [x] I have followed the testing instructions in the [contributing guide](https://github.com/argoproj/argo-helm/blob/master/CONTRIBUTING.md).
* [x] I have signed the CLA and the build is green.
* [x] I will test my changes again once merged to master and published.

Changes are automatically published when merged to `master`. They are not published on branches.

### CODE REVIEW

File: charts/argo/values.yaml [hunk 1]
```diff
@@ -36,6 +36,8 @@ controller:
   image:
     # Overrides .images.tag if defined.
     tag: ""
+  # parallelism dictates how many workflows can be running at the same time
+  parallelism: 10
```
Comment (stefansedich) [inline charts/argo/values.yaml:40]: We should not provide a restrictive default here, can we make the default null and then conditionally set the `paralellism` option in the config-map.

**Review by stefansedich — CHANGES_REQUESTED** [review]
One comment on defaulting.

File: charts/argo/values.yaml (hunk 1 above)
Comment (dbseel) [reply charts/argo/values.yaml]: Done

**Review by stefansedich — APPROVED** [review]
LGTM thanks!

**Pull request merged**